### ISSUE TITLE
When using with pre-commit, version 0.45.0 doesn't work

### ISSUE THREAD

**Issue by arvid-norlander** [igorshubovych/markdownlint-cli#546]
Version 0.45.0 is broken (at least from pre-commit).

```
[INFO] Installing environment for https://github.com/igorshubovych/markdownlint-cli.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/home/user/.cache/pre-commit/repouuagko7n/node_env-default/bin/node', '/home/user/.cache/pre-commit/repouuagko7n/node_env-default/bin/npm', 'install', '--include=dev', '--include=prod', '--ignore-prepublish', '--no-progress', '--no-save')
return code: 1
stdout: (none)
stderr:
    npm warn Unknown cli config "--ignore-prepublish". This will stop working in the next major version of npm.
    npm warn Unknown user config "always-auth". This will stop working in the next major version of npm.
    npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
    npm warn deprecated @humanwhocodes/config-array@0.13.0: Use @eslint/config-array instead
    npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
    npm warn deprecated @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
    npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
    npm warn deprecated eslint@8.57.1: This version is no longer supported. Please see https://eslint.org/version-support for other options.
    npm error code E404
    npm error 404 Not Found - GET https://pkgs.dev.azure.com/which-typed-array/-/which-typed-array-1.1.19.tgz
    npm error 404
    npm error 404  'which-typed-array@https://pkgs.dev.azure.com/which-typed-array/-/which-typed-array-1.1.19.tgz' is not in this registry.
    npm error 404
    npm error 404 Note that you can also install from a
    npm error 404 tarball, folder, http url, or git url.
    npm error A complete log of this run can be found in: /home/user/.npm/_logs/2025-06-04T07_52_12_997Z-debug-0.log
```

Version 0.44.0 works fine. I have no idea what I'm looking at here. I'm a C++ developer I don't know anything about javascript or npm.

The hook in use is (works if changing to 0.44.0):

```yaml
  - repo: https://github.com/igorshubovych/markdownlint-cli
    rev: v0.45.0
    hooks:
      - id: markdownlint-fix
        name: Lint and fix markdown files
        exclude: ^\.azuredevops/.*$
```

Versions:

* pre-commit 4.2.0
* Ubuntu 24.04 LTS

Log files:

* [pre-commit.log](https://github.com/user-attachments/files/20587444/pre-commit.log)
* [2025-06-04T07_56_56_231Z-debug-0.log](https://github.com/user-attachments/files/20587460/2025-06-04T07_56_56_231Z-debug-0.log)

**Comment by DavidAnson** [comment]
It looks like pre-commit is trying to fetch the `which-typed-array` package from `pkgs.dev.azure.com` instead of `registry.npmjs.org` which is what I would expect based on the configuration:

https://github.com/igorshubovych/markdownlint-cli/blob/master/package-lock.json#L8304

If this is something you have overridden in your environment, please undo that and try again. If it's some new behavior from pre-commit, please open an issue for them.

**Comment by DavidAnson** [comment]
I've verified that a properly configured pre-commit still runs this project successfully: https://github.com/DavidAnson/Workflows/actions/runs/15447666848/job/43481570712

**Comment by arvid-norlander** [comment]
Hm, npm is set up to use the company internal mirror, but that is supposed to work transparently, and has done so in the past. I will contact the relevant person internally and ask them what is going on.

**Comment by arvid-norlander** [comment]
After investigating this, it seems that when pre-commit uses npm it strips the path after `https://pkgs.dev.azure.com/`. We haven't figured out why, but it seems to have started somewhat recently, and doesn't depend on the version of pre-commit in use. Pre-commit always uses the latest npm/nodejs apparently, so maybe that is what we need to look into next.
